### PR TITLE
Use `TEST_WS` in all remote-externalities tests

### PR DESCRIPTION
**Pull request by brunopgalvao**
Refactor in accordance with https://github.com/paritytech/polkadot-sdk/issues/2245#issuecomment-1937025951

Prior to this PR, the `remote_tests` test module would either use `TEST_WS` or `DEFAULT_HTTP_ENDPOINT`.

With the PR, `TEST_WS` is the default for the `remote_tests` test module and the fallback is `DEFAULT_HTTP_ENDPOINT`.

The only downside I see to this PR is that for particular tests in the `remote_tests` module, one would want to use a different http endpoint. If that is the case, they would have to manually hardcode the http endpoint for that particular test.

Note: The `TEST_WS` node should fulfill the role for all test cases e.g. include child tries.

Give it a _try_:
```
TEST_WS=wss://rococo-try-runtime-node.parity-chains.parity.io:443 cargo test --features=remote-test -p frame-remote-externalities -- --nocapture
```

### CODE REVIEW

File: substrate/utils/frame/remote-externalities/src/lib.rs [hunk 1]
```diff
@@ -1264,6 +1264,16 @@ mod tests {
...
+	use std::env;
+
+	static TRANSPORT_URL: once_cell::sync::Lazy<String> = once_cell::sync::Lazy::new(|| {
+        env::var("TEST_WS").unwrap_or_else(|_| DEFAULT_HTTP_ENDPOINT.to_string())
+    });
+
+	#[test]
+	fn print_transport_url() {
+		println!("TRANSPORT_URL: {:?}", once_cell::sync::Lazy::<std::string::String>::force(&TRANSPORT_URL));
+	}
```
Comment (liamaharon) [inline substrate/utils/frame/remote-externalities/src/lib.rs:1276]: why's this needed?

**Review by liamaharon — APPROVED** [review]
once_cell maybe overkill for this but fun to use it I guess :) thanks

**Comment by brunopgalvao** [comment]
> once_cell maybe overkill for this but fun to use it I guess :) thanks

It was the only way I could get the `TRANSPORT_URL` to be available in the scope of `mod remote_tests` otherwise `TRANSPORT_URL` would need to be defined for every test.

File: substrate/utils/frame/remote-externalities/src/lib.rs (hunk 1 above)
Comment (brunopgalvao) [reply substrate/utils/frame/remote-externalities/src/lib.rs:1276]: I found that when running the tests, I would look at the output but not know which `TRANSPORT_URL` it was using.

If you think it is not needed, I can remove.

Comment (liamaharon) [reply substrate/utils/frame/remote-externalities/src/lib.rs:1276]: I wonder if it's possible to log in the closure where the static var is defined?

File: substrate/utils/frame/remote-externalities/src/lib.rs [hunk 2]
```diff
@@ -1264,6 +1264,16 @@ mod tests {
 mod remote_tests {
 	use super::test_prelude::*;
 	use std::os::unix::fs::MetadataExt;
+	use std::env;
+
+	static TRANSPORT_URL: once_cell::sync::Lazy<String> = once_cell::sync::Lazy::new(|| {
+        env::var("TEST_WS").unwrap_or_else(|_| DEFAULT_HTTP_ENDPOINT.to_string())
```
Comment (ggwpez) [inline substrate/utils/frame/remote-externalities/src/lib.rs:1270]: You can just use a function:  
```rust
fn endpoint() -> String {
    env::var("TEST_WS").unwrap_or_else(|_| DEFAULT_HTTP_ENDPOINT.to_string())
}
```
the computation is not complicated so no need to make store it.

Comment (brunopgalvao) [reply substrate/utils/frame/remote-externalities/src/lib.rs]: Done. Refactored to use a function instead.

File: substrate/utils/frame/remote-externalities/src/lib.rs (hunk 1 above)
Comment (brunopgalvao) [reply substrate/utils/frame/remote-externalities/src/lib.rs]: That was what I tried to do first but with no avail :(

File: substrate/utils/frame/remote-externalities/src/lib.rs [hunk 3]
```diff
@@ -1266,13 +1266,13 @@ mod remote_tests {
...
-    });
+	fn endpoint() -> String {
+		env::var("TEST_WS").unwrap_or_else(|_| DEFAULT_HTTP_ENDPOINT.to_string())
+	}
 
 	#[test]
 	fn print_transport_url() {
-		println!("TRANSPORT_URL: {:?}", once_cell::sync::Lazy::<std::string::String>::force(&TRANSPORT_URL));
+		println!("TRANSPORT_URL: {:?}", endpoint());
 	}
```
Comment (liamaharon) [inline substrate/utils/frame/remote-externalities/src/lib.rs]: Just's remove this test and put a debug log instead in `fn endpoint` I think

Comment (brunopgalvao) [reply substrate/utils/frame/remote-externalities/src/lib.rs]: If I do smth like this:
```rust
fn endpoint() -> String {
	let endpoint_value = env::var("TEST_WS").unwrap_or_else(|_| DEFAULT_HTTP_ENDPOINT.to_string());
	println!("Endpoint: {:?}", endpoint_value);
	endpoint_value
}
```
I end up with an output like this:
```
running 13 tests
Endpoint: "wss://rococo-try-runtime-node.parity-chains.parity.io:443"
Endpoint: "wss://rococo-try-runtime-node.parity-chains.parity.io:443"
Endpoint: "wss://rococo-try-runtime-node.parity-chains.parity.io:443"
Endpoint: "wss://rococo-try-runtime-node.parity-chains.parity.io:443"
Endpoint: "wss://rococo-try-runtime-node.parity-chains.parity.io:443"
Endpoint: "wss://rococo-try-runtime-node.parity-chains.parity.io:443"
Endpoint: "wss://rococo-try-runtime-node.parity-chains.parity.io:443"
Endpoint: "wss://rococo-try-runtime-node.parity-chains.parity.io:443"
Endpoint: "wss://rococo-try-runtime-node.parity-chains.parity.io:443"
Endpoint: "wss://rococo-try-runtime-node.parity-chains.parity.io:443"
✅ Found 9 keys (1.75s)
✅ Found 0 keys (1.69s)
test remote_tests::can_build_big_pallet ... ok
Endpoint: "wss://rococo-try-runtime-node.parity-chains.parity.io:443"
✅ Found 1 keys (1.70s)
✅ Found 641 keys (2.00s)
[00:00:00] ✅ Downloaded key values 35.9984/s [====================================================================================] 9/9 (0s)
[00:00:00] ✅ Downloaded key values 4.1249/s [=====================================================================================] 1/1 (0s)
✅ Found 641 keys (2.14s)
✅ Found 641 keys (2.25s)
✅ Found 641 keys (2.19s)
░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ 0/641
░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ 0/641
✅ Inserted keys into DB (0.00s)
✅ Found 641 keys (1.99s)
✅ Inserted keys into DB (0.00s)
✅ Found 641 keys (2.00s)
✅ Found 1 keys (1.06s)
[00:00:00] ✅ Downloaded key values 4.0373/s [=====================================================================================] 1/1 (0s)
✅ Inserted keys into DB (0.00s)
[00:00:02] Downloading key values 156.1834/s [==============================================>----------------------------------] 366/641 (2s)
test remote_tests::can_create_child_snapshot ... ok
✅ Loaded snapshot (0.01s)
✅ Loaded snapshot (0.00s)
test tests::can_exclude_from_snapshot ... ok
✅ Loaded snapshot (0.00s)
test tests::can_load_state_snapshot ... ok
[00:00:02] ✅ Downloaded key values 214.2809/s [===============================================================================] 641/641 (0s)
✅ Inserted keys into DB (0.02s)
✅ Loaded snapshot (0.00s)
test remote_tests::snapshot_block_hash_works ... ok
[00:00:02] ✅ Downloaded key values 214.4463/s [===============================================================================] 641/641 (0s)
[00:00:03] ✅ Downloaded key values 211.0836/s [===============================================================================] 641/641 (0s)
✅ Inserted keys into DB (0.01s)
[00:00:03] ✅ Downloaded key values 213.1888/s [===============================================================================] 641/641 (0s)
⠋     0.000 s   Inserting keys into DB...test remote_tests::can_create_snapshot ... ok
[00:00:03] ✅ Downloaded key values 206.8499/s [===============================================================================] 641/641 (0s)
✅ Inserted keys into DB (0.01s)
test remote_tests::can_build_one_small_pallet ... ok
✅ Inserted keys into DB (0.01s)
✅ Inserted keys into DB (0.01s)
✅ Found 641 keys (2.01s)
✅ Found 1 keys (0.99s)
[00:00:04] ✅ Downloaded key values 155.4818/s [===============================================================================] 641/641 (0s)
✅ Inserted keys into DB (0.01s)
[00:00:00] ✅ Downloaded key values 4.0076/s [=====================================================================================] 1/1 (0s)
✅ Loaded snapshot (0.00s)
test remote_tests::offline_else_online_works ... ok
✅ Inserted keys into DB (0.00s)
⠙     1.759 s   Scraping keys...█████████████████████████████████████████████████████████████████████████████████████████████████████████████
Endpoint: "wss://rococo-try-runtime-node.parity-chains.parity.io:443"
✅ Found 495 keys (1.81s)
```
And if I use `debug!("Endpoint: {}", endpoint_value);` instead, then I see no output from that statement in the terminal.

Comment (liamaharon) [reply substrate/utils/frame/remote-externalities/src/lib.rs]: let's just set the default to something sensible and leave the log out

File: substrate/utils/frame/remote-externalities/src/lib.rs [hunk 4]
```diff
@@ -1263,7 +1263,16 @@ mod tests {
...
+	use std::{env, os::unix::fs::MetadataExt};
+
+	fn endpoint() -> String {
+		env::var("TEST_WS").unwrap_or_else(|_| DEFAULT_HTTP_ENDPOINT.to_string())
+	}
+
+	#[test]
+	fn print_transport_url() {
+		println!("TRANSPORT_URL: {:?}", endpoint());
+	}
```
Comment (ggwpez) [inline substrate/utils/frame/remote-externalities/src/lib.rs:1275]:
```suggestion
```
Its not a test.

File: substrate/utils/frame/remote-externalities/src/lib.rs (hunk 3 above)
Comment (brunopgalvao) [reply substrate/utils/frame/remote-externalities/src/lib.rs]: Removed.

File: substrate/utils/frame/remote-externalities/src/lib.rs [hunk 5]
```diff
@@ -1524,13 +1534,10 @@ mod remote_tests {
 
 	#[tokio::test]
 	async fn can_fetch_all() {
-		if std::option_env!("TEST_WS").is_none() {
-			return
-		}
```
Comment (bkchr) [inline substrate/utils/frame/remote-externalities/src/lib.rs:1529]: Did you intentionally enabled them? Before they were disabled if `TEST_WS` wasn't set?

Comment (brunopgalvao) [reply substrate/utils/frame/remote-externalities/src/lib.rs]: Good point. 

It makes sense to **only** use `TEST_WS` for larger tests vs defaulting to `DEFAULT_HTTP_ENDPOINT` which is `https://rpc.polkadot.io:443`

Added these conditionals back in.

**Review by bkchr — APPROVED** [review]

**Pull request merged**